deposit/src/main/java/edu/unc/lib/deposit/fcrepo4/RegisterToLongleafJob.java

```diff
@@ -0,0 +1,166 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package edu.unc.lib.deposit.fcrepo4;
+
+import edu.unc.lib.deposit.work.AbstractDepositJob;
+import edu.unc.lib.dl.rdf.CdrDeposit;
+import org.apache.jena.rdf.model.Model;
+import org.apache.jena.rdf.model.Bag;
+import org.apache.jena.rdf.model.Resource;
+import org.apache.jena.rdf.model.Statement;
+import org.apache.jena.rdf.model.NodeIterator;
+import org.apache.jena.rdf.model.StmtIterator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URI;
+import java.nio.file.Paths;
+
+/**
+ * Job which registers ingested files to longleaf
+ *
+ * @author smithjp
+ *
+ */
+public class RegisterToLongleafJob extends AbstractDepositJob {
+    private static final Logger log = LoggerFactory.getLogger(RegisterToLongleafJob.class);
+    private static final Logger longleafLog = LoggerFactory.getLogger("longleaf");
+
+    private String longleafBaseCommand;
+
+    public RegisterToLongleafJob() {
+        super();
+    }
+
+    public RegisterToLongleafJob(String uuid, String depositUUID) {
+        super(uuid, depositUUID);
+    }
+
+    @Override
+    public void runJob() {
+        log.info("Registering files from deposit {} to longleaf", getDepositPID());
+
+        Model model = getReadOnlyModel();
+        Bag depositBag = model.getBag(getDepositPID().getRepositoryPath());
+
+        registerFilesToLongleaf(depositBag);
+    }
+
+    /**
+     * Iterates through deposit bag to find files which can be registered with
+     * longleaf
+     *
+     * @param resc
+     */
+    public void registerFilesToLongleaf(Resource resc) {
+        StmtIterator statementIterator = resc.listProperties();
+        try {
+            while (statementIterator.hasNext()) {
+                Statement currentStatement = statementIterator.nextStatement();
+
+                // find storageUri, descriptiveStorageUri, fitsStorageUri, premisStorageUri
+                if (currentStatement.getPredicate().toString().matches(".*Uri")) {
+                    URI storageUri = URI.create(currentStatement.getString());
+                    if (currentStatement.getString().matches(".*original_file")) {
+                        String checksum = resc.getProperty(CdrDeposit.md5sum).getString();
+                        registerFile(storageUri, checksum);
+                    } else {
+                        registerFile(storageUri, "");
```

Comment:
one minor suggestion, i'd recommend using `null` to represent the absence of something rather than an empty string